include/spdlog/pattern_formatter-inl.h

```diff
@@ -1021,6 +1021,7 @@ SPDLOG_INLINE pattern_formatter::pattern_formatter(
     , pattern_time_type_(time_type)
     , last_log_secs_(0)
     , custom_handlers_(std::move(custom_user_flags))
+    , needs_time_(false)
```

Comment:
please rename to “need_localtime_”
Some of the flags needs time but not localtime.